### ISSUE TITLE
\DeclareFieldFormat breaks with long argument

### ISSUE THREAD

**Issue by moewew** [plk/biblatex#1037]
From https://tex.stackexchange.com/q/559645/35864.

https://github.com/plk/biblatex/commit/089ce33f1e3cbde9433fda6e8ab4c82e96018b80 breaks long arguments in `\DeclareFieldFormat` because it explicitly grabs the argument. 

That breaks code like
```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=authoryear, backend=biber]{biblatex}

\DeclareFieldFormat{annotation}{\par#1}

\addbibresource{biblatex-examples.bib}


\begin{document}
\cite{sigfridsson}
\printbibliography
\end{document}
```

A **temporary workaround** is to make `\blx@defformat@i` long. Thanks to the new file hooks it should be possible to inject the necessary code when style files are loaded.
```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\makeatletter
\csdef{blx@filehook@preload@chicago-notes.cbx}{%
\long\def\blx@defformat@i##1##2##3[##4]##5{%
  \notblank{##3}
    {\blx@resetformat{##2}{##5}}
    {}%
  \def\blx@defformat@a{##2}%
  \def\blx@defformat@b{##4}%
  \blx@xsanitizeafter{\def\blx@defformat@c}{##5}%
  \afterassignment\blx@defformat@ii
  ##1}}
\makeatother

\usepackage{biblatex-chicago}

\addbibresource{biblatex-examples.bib}


\begin{document}
\cite{sigfridsson}
\printbibliography
\end{document}
```

**Comment by moewew** [comment]
@plk Seems we will need another bugfix release soon-ish (see #1039). Since that one has a simpler temporary workaround and only occurs with one of the more exoctic commands (`\volcite`), I'd like to wait another few days, maybe until the end of the week to see if there are some other bugs.

I noticed that the 3.15a documentation on CTAN still only says 3.15 on the title page. I checked with the `.zip`s on SourceForge and indeed, the `biblatex.pdf` `biblatex-3.15a.tds.tgz` has 3.15a on the title page, but the one in `biblatex-3.15a.tgz` only has 3.15. Seems as though the non-TDS `.zip` is built from the commit before the version is stepped...
